lib/client/workflow-client/index.js

```diff
@@ -0,0 +1,299 @@
+ */
+function stop() {
+  var donePromise = mediator.promise(workflowSyncSubscribers.getTopic(CONSTANTS.TOPICS.STOP, CONSTANTS.DONE_PREFIX));
+
+  var errorPromise = mediator.promise(workflowSyncSubscribers.getTopic(CONSTANTS.TOPICS.STOP, CONSTANTS.ERROR_PREFIX));
+
+  mediator.publish(workflowSyncSubscribers.getTopic(CONSTANTS.TOPICS.STOP));
+
+  return getTopicPromises(donePromise, errorPromise);
+}
+
+/**
+ * Forcing the workflows to sync to the remote store.
+ */
+function forceSync() {
+  var donePromise = mediator.promise(workflowSyncSubscribers.getTopic(CONSTANTS.TOPICS.FORCE_SYNC, CONSTANTS.DONE_PREFIX));
+
+  var errorPromise = mediator.promise(workflowSyncSubscribers.getTopic(CONSTANTS.TOPICS.FORCE_SYNC, CONSTANTS.ERROR_PREFIX));
+
+  mediator.publish(workflowSyncSubscribers.getTopic(CONSTANTS.TOPICS.FORCE_SYNC));
+
+  return getTopicPromises(donePromise, errorPromise);
+}
+
+/**
+ * Safe stop forces a synchronisation to the remote server and then stops the synchronisation process.
+ * @returns {Promise}
+ */
+function safeStop() {
+  return forceSync().then(stop);
+}
+
+
+/**
+ * Waiting for the synchronisation process to complete to the remote cluster.
+ */
+function waitForSync() {
+  return mediator.promise(workflowSyncSubscribers.getTopic(CONSTANTS.TOPICS.SYNC_COMPLETE));
+}
+
+function ManagerWrapper(_manager) {
+  this.manager = _manager;
+  var self = this;
+
+  var methodNames = ['create', 'read', 'update', 'delete', 'list', 'start', 'stop', 'safeStop', 'forceSync', 'waitForSync'];
+  methodNames.forEach(function(methodName) {
+    self[methodName] = function() {
+      return q.when(self.manager[methodName].apply(self.manager, arguments));
+    };
+  });
+
+  self.checkStatus = function() {
+    return self.manager.checkStatus.apply(self.manager, arguments);
+  };
+
+  self.stepReview = function() {
+    return self.manager.stepReview.apply(self.manager, arguments);
+  };
+
+  self.nextStepIndex = function() {
+    return self.manager.nextStepIndex.apply(self.manager, arguments);
+  };
+}
+
+/**
+ *
+ * @param steps
+ * @param result
+ * @returns {{nextStepIndex: number, complete: *}}
+ */
+function stepReview(steps, result) {
+  var stepIndex = -1;
+  var complete;
+  if (result && result.stepResults && result.stepResults.length !== 0) {
+    complete = true;
+    for (var i=0; i < steps.length; i++) {
+      var step = steps[i];
+      var stepResult = result.stepResults[step.code];
+      if (stepResult && (stepResult.status === 'complete' || stepResult.status === 'pending')) {
+        stepIndex = i;
+        if (stepResult.status === 'pending') {
+          complete = false;
+        }
+      } else {
+        break;
+      }
+    }
+  }
+  return {
+    nextStepIndex: stepIndex,
+    complete: complete // false is any steps are "pending"
+  };
+}
+
+function nextStepIndex(steps, result) {
+  return this.stepReview(steps, result).nextStepIndex;
+}
+
+/**
+ *
+ * Checking the status of a workorder
+ *
+ * TODO: Constants
```

Comment:
:] See my comment bellow. It feels that we can move this to constants without much effort and remove this TODO :)